### PR TITLE
Update pandas_extensions.py

### PR DESCRIPTION
**Pull request by 149189**
--- PR TEMPLATE INSTRUCTIONS (1) ---

Looking to submit a Hamilton Dataflow to the sf-hamilton-contrib module? If so go the the `Preview` tab and select the appropriate sub-template:
* [sf-hamilton-contrib template](?expand=1&template=HAMILTON_CONTRIB_PR_TEMPLATE.md)

Else, if not, please remove this block of text.

--- PR TEMPLATE INSTRUCTIONS (2) ---

[Short description explaining the high-level reason for the pull request]

## Changes

1. Added `PandasTableReader` class in `pandas_extensions.py` for reading tabular data using `pandas.read_table`.
2. Implemented necessary methods to facilitate the loading of data.
3. Registered the `PandasTableReader` class in the data loader registry.

**Functionality:**

The `PandasTableReader` class allows reading tabular data with flexible loading options using `pandas.read_table`. It inherits from the `DataLoader` class and provides an easy way to load data into a DataFrame.

**Documentation:**

Updated the `data_loader` module documentation to include information about the new `PandasTableReader` class and its usage for loading tabular data.

**Reviewer Note:**

Please review the changes made in `pandas_extensions.py` and the associated tests in `test_pandas_extensions.py`. Feedback and suggestions are welcome.

## How I tested this

Added a unit test (`test_read_table`) in `test_pandas_extensions.py` to validate the functionality of the `PandasTableReader` class. The test covers loading tabular data and comparing the expected DataFrame with the loaded DataFrame.

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by 149189** [comment]
@skrawcz Hey! I have made some Changes can you review my code please i am not understanding why the isort failed

File: hamilton/plugins/pandas_extensions.py [hunk 1]
```diff
@@ -6,6 +6,16 @@
 from io import BufferedReader, BytesIO, StringIO
 from pathlib import Path
 from typing import Any, Callable, Collection, Dict, Iterator, List, Optional, Tuple, Type, Union
+import pandas as pd
+from io import BytesIO, BufferedReader
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:10]: this is a redefinition of the above. remove

File: hamilton/plugins/pandas_extensions.py [hunk 2]
```diff
@@ -6,6 +6,16 @@
 from io import BufferedReader, BytesIO, StringIO
 from pathlib import Path
 from typing import Any, Callable, Collection, Dict, Iterator, List, Optional, Tuple, Type, Union
+import pandas as pd
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:9]: this is already done below. remove.

File: hamilton/plugins/pandas_extensions.py [hunk 3]
```diff
@@ -6,6 +6,16 @@
...
+from io import BytesIO, BufferedReader
+from typing import Union, Dict, Any, Type, Tuple, Collection, Optional
+from pandas import DataFrame
+from pandas.api.types import infer_dtype
+from pandas.io.common import get_handle
+from pandas.io.parsers import read_table
+from pandas.io.sql import SQLTable, pandasSQL_builder
+from pandas.io.sql import get_schema
+from pathlib import Path
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:18]: these all look like redefinitions and not required.

File: hamilton/plugins/pandas_extensions.py [hunk 4]
```diff
@@ -983,6 +993,165 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
     @classmethod
     def name(cls) -> str:
         return "feather"
+    
+class PandasTableReader(DataLoader):
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:997]: This should be decorated with a dataclass, and look like the other classes.

File: hamilton/plugins/pandas_extensions.py [hunk 5]
```diff
@@ -983,6 +993,165 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
...
+            "storage_options": self.storage_options,
+        }
+        return kwargs
+
+    def load_data(self, type: Type) -> Tuple[DataFrame, Dict[str, Any]]:
+        filepath_or_buffer = get_filepath_or_buffer(
+            self.path_or_buffer
+        )
+        kwargs = self._get_loading_kwargs()
+        data = read_table(filepath_or_buffer, **kwargs)
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1153]: this should be pd.read_table, not importing it directly.

File: hamilton/plugins/pandas_extensions.py [hunk 6]
```diff
@@ -983,6 +993,165 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
...
+        return data, {}
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1154]: this should use the metadata utility like the others are using to provide metadata on the file being loaded.

**Review by skrawcz — CHANGES_REQUESTED** [review]
Thanks for putting this up. To merge this, we need a few changes. See comments.

**Comment by 149189** [comment]
@skrawcz Bro I made all the changes you told and the branch has been merged. Thanks for this experience this was my first contribution!!

**Comment by skrawcz** [comment]
@149189 are you still working on this?

**Comment by skrawcz** [comment]
closing because I haven't heard back.

**Pull request closed without merging**